SeaPublicWebsite/Controllers/CookieController.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using GovUkDesignSystem.GovUkDesignSystemComponents;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using SeaPublicWebsite.Helpers;
+using SeaPublicWebsite.Models.Cookies;
+using SeaPublicWebsite.Services;
+using SeaPublicWebsite.Services.Cookies;
+
+namespace SeaPublicWebsite.Controllers;
+
+public class CookieController: Controller
+{
+    private readonly CookieService cookieService;
+    private readonly GlobalConstants globalConstants;
+
+    public CookieController(CookieService cookieService, GlobalConstants globalConstants)
+    {
+        this.cookieService = cookieService;
+        this.globalConstants = globalConstants;
+    }
+
+    [HttpGet("/cookies")]
+    public IActionResult CookieSettings_Get(bool changesHaveBeenSaved = false)
+    {
+        cookieService.TryGetCookie<CookieSettings>(Request, cookieService.Configuration.CookieSettingsCookieName, out var cookie);
+        
+        var viewModel = new CookieSettingsViewModel
+        {
+            GoogleAnalytics = cookie?.GoogleAnalytics is true,
+            ChangesHaveBeenSaved = changesHaveBeenSaved
+        };
+        TempData["BannerState"] = BannerState.Hide;
+        return View("CookieSettings", viewModel);
+    }
+
+    [HttpPost("/cookies")]
+    [ValidateAntiForgeryToken]
+    public IActionResult CookieSettings_Post(CookieSettingsViewModel viewModel)
+    {
+        var cookieSettings = new CookieSettings
+        {
+            Version = cookieService.Configuration.CurrentCookieMessageVersion,
+            GoogleAnalytics = viewModel.GoogleAnalytics,
+        };
+        cookieService.SetCookie(Response, cookieService.Configuration.CookieSettingsCookieName, cookieSettings);
+        return CookieSettings_Get(changesHaveBeenSaved: true);
+    }
+
+    [HttpPost("/cookie-consent")]
+    [ValidateAntiForgeryToken]
+    public IActionResult CookieConsent(CookieConsent cookieConsent)
+    {
+        if (cookieConsent.Consent == "hide")
+        {
+            TempData["BannerState"] = BannerState.Hide;
```

Comment:
Because we are redirecting, we cannot store the banner state in a ViewBag since it would get reset. TempData is preserved through the chain of redirects.

Reply:
Does this work at the moment? I believe that `TempData` relies on there being a `session` which we don't currently have (and would like to avoid if possible)

Reply:
I suspect that this might look like it's working because of the logic in `_Layout.cshtml`

Reply:
If I'm correct, then please replace the references to `TempData` with `ViewBag`

Reply:
ViewBag still doesn't work because we would redirect. However I just got an idea, instead of storing whether the user saw the banner or not in the cookie, we can store what banner we should display. I'll see if I can get that working